File: realdsa1/my_script.py
import os
import sys
from collections import deque

def bfs_directory(start_path, target):
    queue = deque([start_path])
    results = []

    while queue:
        current = queue.popleft()

        for entry in os.listdir(current):
            full = os.path.join(current, entry)

            if os.path.isdir(full):
                queue.append(full)

            elif entry.endswith(".txt"):
                # linear search in file
                with open(full, "r", encoding="utf8", errors="ignore") as f:
                    for line in f:
                        if "Number of people:" in line:
                            # extract digits manually
                            num = ''.join(ch for ch in line if ch.isdigit())

                            if num == target:
                                year = os.path.basename(os.path.dirname(full))
                                month = entry.replace(".txt", "")
                                results.append((month, year))

    return results


# main execution
if len(sys.argv) < 2:
    print("Usage: python bfs_search.py <value>")
    sys.exit(1)

target = sys.argv[1]
matches = bfs_directory("data", target)

if not matches:
    print("No matches found.")
else:
    for m, y in matches:
        print(f"{m}, {y}")

File: realdsa1/my_script.js
const fs = require("fs");
const path = require("path");

// BFS through folders
function bfsDirectory(startPath, target) {
    const queue = [startPath];
    const results = [];

    while (queue.length > 0) {
        const current = queue.shift(); // BFS pop

        const entries = fs.readdirSync(current);

        for (const entry of entries) {
            const fullPath = path.join(current, entry);
            const stat = fs.statSync(fullPath);

            if (stat.isDirectory()) {
                // Add directory to BFS queue
                queue.push(fullPath);

            } else if (entry.endsWith(".txt")) {
                // linear search inside each file
                const content = fs.readFileSync(fullPath, "utf8");
                const lines = content.split("\n");

                for (let line of lines) {
                    if (line.includes("Number of people:")) {

                        // extract digits manually
                        const num = line.split(":")[1].replace(/[^0-9]/g, "");

                        if (num === target) {
                            const year = path.basename(path.dirname(fullPath));
                            const month = entry.replace(".txt", "");
                            results.push({ month, year });
                        }
                    }
                }
            }
        }
    }

    return results;
}


// main execution
const target = process.argv[2];
if (!target) {
    console.log("Usage: node bfs_search.js <value>");
    process.exit(1);
}

const dataPath = path.join(__dirname, "data");
const matches = bfsDirectory(dataPath, target);

if (matches.length === 0) {
    console.log("No matches found.");
} else {
    matches.forEach(m => console.log(`${m.month}, ${m.year}`));
}

